### PR TITLE
Chore/update to next 15

### PR DESCRIPTION
**Pull request by saltcod**
update our docs to reflect https://github.com/vercel/next.js/pull/71631

### CODE REVIEW

File: apps/docs/content/guides/auth/passwords.mdx [hunk 1]
```diff
@@ -184,7 +184,7 @@ export async function GET(request: NextRequest) {
   redirectTo.pathname = next
 
   if (token_hash && type) {
-    const supabase = createClient()
```
Comment (ivasilov) [inline apps/docs/content/guides/auth/passwords.mdx:187]: [this page](https://docs-git-chore-update-to-next-15-supabase.vercel.app/docs/guides/auth/passwords?queryGroups=flow&flow=pkce) has another `const supabase = createClient()`

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 2]
```diff
@@ -341,9 +341,9 @@ Most web apps and websites can utilize Google's [personalized sign-in buttons](h
    // Use 'nonce' when invoking the supabase.auth.signInWithIdToken() method
    ```
 
-### One-tap with NextJS
```
Comment (ivasilov) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:344]: [This page](https://docs-git-chore-update-to-next-15-supabase.vercel.app/docs/guides/auth/social-login/auth-google?queryGroups=environment&environment=server&queryGroups=framework&framework=nextjs) has another `createClient` which needs await.

File: apps/docs/content/guides/auth/server-side/nextjs.mdx [hunk 3]
```diff
@@ -726,9 +726,9 @@ import { useState } from 'react'
 
 import { createClient } from '@/utils/supabase/component'
 
-export default function LoginPage() {
+export default async function LoginPage() {
```
Comment (charislam) [inline apps/docs/content/guides/auth/server-side/nextjs.mdx:729]:
```suggestion
export default function LoginPage() {
```

I think this one is Pages Router so should still be synchronous?

File: apps/docs/content/guides/auth/server-side/nextjs.mdx [hunk 4]
```diff
@@ -726,9 +726,9 @@ import { useState } from 'react'
...
   const router = useRouter()
-  const supabase = createClient()
+  const supabase = await createClient()
```
Comment (charislam) [inline apps/docs/content/guides/auth/server-side/nextjs.mdx:731]:
```suggestion
  const supabase = createClient()
```

pages router

File: apps/docs/content/guides/auth/server-side/nextjs.mdx [hunk 5]
```diff
@@ -940,7 +940,7 @@ export default function PublicPage({ data }: { data?: any[] }) {
 }
 
 export async function getStaticProps() {
-  const supabase = createClient()
+  const supabase = await createClient()
```
Comment (charislam) [inline apps/docs/content/guides/auth/server-side/nextjs.mdx:943]:
```suggestion
  const supabase = createClient()
```

pages router

**Review by charislam — APPROVED** [review]

**Comment by saltcod** [comment]
checked studio and docs, look good.

**Pull request merged**